src/tasks/setApprovals.ts

```diff
@@ -0,0 +1,116 @@
+import { promises as fs } from "fs";
+
+import "@nomiclabs/hardhat-ethers";
+import { Contract } from "ethers";
+import { task, types } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { SettlementEncoder } from "../ts";
+
+import { createGasEstimator, gweiToWei } from "./ts/gas";
+import { prompt } from "./ts/tui";
+
+interface Approval {
+  spender: string;
+  token: string;
+}
+
+interface Args {
+  input: string;
+  dryRun: boolean;
+  gasInGwei: number;
+}
+
+async function setApprovals(
+  { input, dryRun, gasInGwei }: Args,
+  hre: HardhatRuntimeEnvironment,
+) {
+  // Instantiate Settlement Contract
+  const settlementDeployment = await hre.deployments.get("GPv2Settlement");
+  const settlement = new Contract(
+    settlementDeployment.address,
+    settlementDeployment.abi,
+  ).connect(hre.ethers.provider);
+  const [signer] = await hre.ethers.getSigners();
+
+  //Instantiate ERC20 ABI
+  const IERC20 = await hre.artifacts.readArtifact(
+    "src/contracts/interfaces/IERC20.sol:IERC20",
+  );
+  const token = new Contract(
+    hre.ethers.constants.AddressZero,
+    IERC20.abi,
+    hre.ethers.provider,
+  );
+
+  // Load approval list and encode interaction for each entry
+  const approvals: Approval[] = JSON.parse(await fs.readFile(input, "utf-8"));
+  const encoder = new SettlementEncoder({});
+  approvals.forEach((approval) => {
+    encoder.encodeInteraction({
+      target: approval.token,
+      callData: token.interface.encodeFunctionData("approve", [
+        approval.spender,
+        hre.ethers.constants.MaxUint256,
+      ]),
+    });
+  });
+  const finalSettlement = encoder.encodedSettlement({});
+  const gasEstimator = createGasEstimator(hre, {
+    blockNative: true,
+  });
```

Comment:
I would make this a flag. The default `ethers` gas price estimator should be good enough (I think it just does 2x base gas fee + some small priority fee).

Reply:
I think for these types of interactions you'd usually set your gas limit manually (you are not really in a rush to get this tx mined), so instead of inflating the cmd arguments, I'll just default it to false.

Reply:
I think for these types of interactions you'd usually set your gas limit manually (you are not really in a rush to get this tx mined), so instead of inflating the cmd arguments, I'll just default it to false.

Reply:
Sounds good to me.